client/session.go

```diff
@@ -153,18 +153,27 @@ func (s *session) Open() error {
 	s.seriesIteratorsPool.Init()
 	s.Unlock()
 
-	currTopologyMap := s.topo.GetAndSubscribe(s.topoMapCh)
+	currTopologyMap, topologyWatch, err := s.topo.GetAndSubscribe()
+	if err != nil {
+		return err
```

Comment:
That would mean two consumers could concurrently call open which we're trying to prevent here unfortunately :(